source/material_model/reactive_fluid_transport.cc

```diff
@@ -377,6 +392,10 @@ namespace aspect
                              "use polynomials to describe hydration and dehydration reactions for four different "
                              "rock compositions as defined in Tian et al., 2019, or the Katz et. al. 2003 mantle "
                              "melting model.");
```

Comment:
This material model now has a lot of parameters that are not being used when the katz2003 model is selected. The declaration will be overwritten by the ones in `ReactionModel::Katz2003MantleMelting<dim>::declare_parameters(prm)`. For example, the default value of the "Reference permeability" will not be used and instead be overwritten by the one from the reaction model. There are two potential solutions

(1) If the computation of the melt outputs is similar, you remove the original parameters here, always call `katz2003_model.calculate_fluid_outputs(in, out, reference_T)`, and make any small modifications necessary afterwards. That would be nice because it would also reduce the lines of code. 
(2) If that does not work well, you could move the declare and parse parameters function of the ReactionModel::Katz2003MantleMelting into its own section in the parameter file, something like

```
prm.enter_subsection("Reactive Fluid Transport Model");
        {
          prm.enter_subsection("Katz 2003 model");
            {
```
This way, they would be separate.

Reply:
I think I'm leaning towards the second solution because the Katz2003 model might not be the only melting model ReactiveFluidTransport has, I think I'll end up adding other melting models after the hackathon. Then it might make more sense for each melt model to have its own parameter subsection. Does that sound ok?